lib/metatron/templates/concerns/pod_producer.rb

```diff
@@ -45,6 +46,16 @@ def formatted_security_context
 
         def formatted_tolerations = tolerations&.any? ? { tolerations: } : {}
         def formatted_volumes = volumes&.any? ? { volumes: } : {}
+
+        def volume_claim_templates
+          if persistent_volume_claims&.any?
+            {
+              volumeClaimTemplates: persistent_volume_claims
```

Comment:
I think, since `persistent_volume_claims` will be an Array of `Metatron::Templates::PersistentVolumeClaims`, you'll need to change this to call `render` on these:

```suggestion
              volumeClaimTemplates: persistent_volume_claims.map(&:render)
```

You could also try making it even smarter to only call `render` if the member of the Array `responds_to?(:render)`, otherwise pass it through as-is (so you can support Hashes here as well), but this would be good enough for now.

Reply:
done!

Reply:
might be a cool idea to make the base render method a bit smarter by crawling through the underlying structure, checking if anything can be rendered, and rendering them if need be.

Reply:
You're not wrong. This would be a great idea!